### PR TITLE
[2.0] NUKE and much-needed CI work

### PR DESCRIPTION
**Pull request by Perksey**
# Summary of the PR
A short summary of this PR and what it adds/removes/fixes.

# Related issues, Discord discussions, or proposals
Links go here.

# Further Comments

### CODE REVIEW

**Comment by Perksey** [comment]
@HurricanKai This is ready for review.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -10,25 +10,22 @@ on:
 
 jobs:
   Build:
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```
Comment (HurricanKai) [inline .github/workflows/build.yml:13]: Does this have to be windows?

File: build/nuke/Build.cs [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+            {
+                return false;
+            }
+            
+            try
+            {
+                MSBuildToolPathResolver.Resolve(); // throws if it can't find msbuild
+                return true;
+            }
+            catch (Exception ex)
```
Comment (HurricanKai) [inline build/nuke/Build.cs:57]:
```suggestion
            catch
```

File: build/nuke/Build.cs [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+    [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
+    readonly string Configuration = IsLocalBuild ? "Debug" : "Release";
+
+    [Parameter("The API key used to push packages and symbols packages to NuGet")] readonly string NugetApiKey;
+    [Parameter("NuGet -NoServiceEndpoint")] readonly bool NugetNoServiceEndpoint;
+    [Parameter("NuGet feed")] readonly string NugetFeed = "https://api.nuget.org/v3/index.json";
+    [Parameter("NuGet username")] readonly string NugetUsername;
+    [Parameter("NuGet password")] readonly string NugetPassword;
+    [Parameter("Extra properties passed to MSBuild commands")] readonly string[] MsbuildProperties =
+        Array.Empty<string>();
```
Comment (HurricanKai) [inline build/nuke/Build.cs:83]: Nit: Style

File: src/Core/Silk.NET.NUKE.SshAgent/Program.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
+﻿using System;
```
Comment (HurricanKai) [inline src/Core/Silk.NET.NUKE.SshAgent/Program.cs:1]: License

**Review by HurricanKai — CHANGES_REQUESTED** [review]
Where is BuildTools in this?
I'm assuming this works fine in IDE.

File: build/nuke/Build.cs (hunk 3 above)
Comment (Perksey) [reply build/nuke/Build.cs:83]:
```suggestion
    [Parameter("Extra properties passed to MSBuild commands")]
    readonly string[] MsbuildProperties = Array.Empty<string>(); 
```

File: src/Core/Silk.NET.NUKE.SshAgent/Program.cs (hunk 4 above)
Comment (Perksey) [reply src/Core/Silk.NET.NUKE.SshAgent/Program.cs:1]: Good catch.
```suggestion
// This file is part of Silk.NET.
// 
// You may modify and distribute Silk.NET under the terms
// of the MIT license. See the LICENSE file for details.

﻿using System;
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (Perksey) [reply .github/workflows/build.yml:13]: Yes, because Desktop MSBuild is required for the Android and iOS feature sets.

**Comment by Perksey** [comment]
BuildTools is used in the RegenerateBindings task, called by FullPack and FullPushToNuGet. The Full tasks aren't used by CI/CD because to ensure that builds are deterministic, I only want to build from the bindings already in the repo.

**Review by HurricanKai — APPROVED** [review]

**Pull request merged**